src/main/java/de/rabea/gui/JavaFXUi.java

```diff
@@ -18,7 +18,9 @@ public void setGameRunner(GameRunner gameRunner) {
 
     @Override
     public void displayBoard(Board board, Player player) {
-        viewUpdater.showBoard((GuiPlayer) player, board, gameRunner, false);
+        if (!player.hasMove()) {
+            viewUpdater.showBoard((GuiPlayer) player, board, gameRunner, false);
```

Comment:
The issue is that I specifically need a GuiPlayer here because it gets passed to a click handler which calls `addMove()` on the GuiPlayer when a click happens. No other player has this method because they all get their moves in a different way. 

I could change it to use the Player abstract class instead and add the `addMove()` method to the class. But none of the other players use it, so it will just be empty... I don't think that's great. What would you suggest here?